datasets/fr/assemblee/crawler.py

```diff
@@ -0,0 +1,181 @@
+        h.apply_name(
+            collabo,
+            prefix=prefix,
+            first_name=first_name,
+            last_name=last_name,
+        )
+        collabo.set("topics", "role.rca")
+        yield collabo
+
+        link = context.make("Associate")
+        link.id = context.make_slug(uid, f"associate", prefix, first_name, last_name)
+        link.set("person", person)
+        link.set("associate", collabo)
+        link.set("relationship", "collaborateur")
+        start_date = c.pop("dateDebut")
+        if is_not_nil(start_date):
+            link.set("startDate", start_date)
+        end_date = c.pop("dateFin")
+        if is_not_nil(end_date):
+            link.set("endDate", end_date)
+        yield link
+        context.audit_data(c)
+
+
+def crawl_acteur(context, data: Dict[str, Any]):
+    """Extract MNA information from JSON."""
+    acteur = data.pop("acteur")
+    context.audit_data(data, ["@xmlns", "profession"])
+
+    person = context.make("Person")
+    # Member UID
+    uid = acteur.pop("uid")
+    uid_text = uid.pop("#text")
+    context.log.debug(f"Unique ID is {uid_text}")
+    person.id = context.make_slug(uid_text)
+    context.audit_data(uid, ["@xmlns:xsi", "@xsi:type"])
+
+    # Name DOB, POB, DOD (or DØD if you're Danish)
+    ec = acteur.pop("etatCivil")
+    ident = ec.pop("ident")
+    given_name = ident.pop("prenom")
+    family_name = ident.pop("nom")
+    h.apply_name(
+        person,
+        prefix=ident.pop("civ"),
+        first_name=given_name,
+        last_name=family_name,
+        lang="fra",
+    )
+    context.audit_data(ident, ["alpha", "trigramme"])
+    birth = ec.pop("infoNaissance")
+    dob = birth.pop("dateNais")
+    if is_not_nil(dob):
+        person.set("birthDate", h.parse_date(dob, ["%Y-%m-%d"]))
+    city = birth.pop("villeNais")
+    if is_not_nil(city):
+        person.set("birthPlace", city)
+    country = birth.pop("paysNais")
+    if is_not_nil(country):
+        person.set("birthCountry", country)
+    # Ignore birth departement for now
+    context.audit_data(birth, ["depNais"])
+    dod = ec.pop("dateDeces")
+    if is_not_nil(dod):
+        person.set("deathDate", h.parse_date(dod, ["%Y-%m-%d"]))
+    context.audit_data(ec)
+
+    # We'll include this URL in the data as it's quite useful
+    hatvp = acteur.pop("uri_hatvp")
+    if is_not_nil(hatvp):
+        person.set("sourceUrl", hatvp)
+
+    # Addresses and phone numbers are available because transparence!
+    # But we do not include them because vie privée!
+    acteur.pop("adresses")
+
+    # Now make position etc
+    position = h.make_position(
+        context,
+        "Member of the French National Assembly",
+        country="fr",
+        topics=["gov.national", "gov.legislative"],
+    )
+    categorisation = categorise(context, position, is_pep=True)
+    if not categorisation.is_pep:
+        context.log.warning(f"Member {given_name} {family_name} is not PEP")
+        return
```

Comment:
I like this. I think the need for this check is a bit of a wart on the design of categorisation, but given that, this is a bit more helpful than just existing with a "no entities exported" error with no explanation at the end when the crawler knows that it ought to be a PEP.

Reply:
Yes - perhaps it should be `info` instead of `warning` since I guess it is valid (but confusing) that you could pass `is_pep=True` to `categorize` and still get `False`?